### DISCUSSION TITLE
Office hours v0.1 (14thJune)

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/LinkFree#7576] [General]
Topics to discuss on the call

⏰ 14th June 1pm UTC

1. Profile + User relationship https://github.com/EddieHubCommunity/LinkFree/issues/7065
2. Logging https://github.com/EddieHubCommunity/LinkFree/pull/7562
3. Questions
4. Ideas

Please start a new thread for each topic discussion (yes create multiple threads)

**Comment by eddiejaoude** [comment]
Please feedback on how the call went and any suggestions for the future? 👇

**Reply by ikrishg** [reply]
It was awesome! 💪🏻 I think the next time we should also include the Knight Commander General of EddieHub, Sara 🔥

**Reply by CBID2** [reply]
I second this! :)

**Comment by eddiejaoude** [comment]
Logging discussion

- Log rotation
   - I don't think it is required for Vercel
- Export to centralised location
   - Gives access to maintainers

**Reply by CBID2** [reply]
If possible, I can do the docs on the logging process. For context, I’m referring to the [idea 4](https://github.com/EddieHubCommunity/LinkFree/issues/7437)

**Comment by CBID2** [comment]
Troubleshooting guide creation discussion.
Would Linkfree benefit from having a troubleshooting guide? I noticed that there really isn’t any.

**Reply by eddiejaoude** [reply]
For example an FAQ?

- Q. I have no data in my database
- A. Visit url ... to load data

**Reply by eddiejaoude** [reply]
> I also mean like if certain things are not popping up on their Linkfree profile

For example FAQ?

- Q. How do I get testimonials on my profile
- A.

**Reply by ikrishg** [reply]
Have we ever thought about moving linkfree docs to docs.linkfree.io? We can have more extensive navigation and lots of navigation

**Reply by CBID2** [reply]
I raised an issue about this. Check #7580 when you can @eddiejaoude.

**Reply by SaraJaoude** [reply]
There is an FAQ page on the website already: https://linkfree.io/docs/faqs

**Comment by eddiejaoude** [comment]
Integrate with CrowdSec (export logs and also use middleware to protect) https://www.crowdsec.net

**Comment by ikrishg** [comment]
# Integration with Axiom (monitoring)

- Free uptil 500GB of logs/month
- Stores data upto one month of 100Gb worth of data (whichever is reached first) for free
- 30 query hours included in free plan
- Will sent automatic discord and email notifications if needed

**SOLID ALTERNATIVE TO WHAT WE PAY 50$ TO VERCEL FOR**


Here's a sample of a custom dashboard:
![image](https://github.com/EddieHubCommunity/LinkFree/assets/135469703/1ba513f1-5012-42ae-ab12-55d7d000587f)

Here's the default one you get by connecting axiom and vercel:

![image](https://github.com/EddieHubCommunity/LinkFree/assets/135469703/04783482-1a37-42e4-9e6b-d6414545c4de)
![image](https://github.com/EddieHubCommunity/LinkFree/assets/135469703/5c744f2c-e165-4178-82bc-1b29b8901986)
![image](https://github.com/EddieHubCommunity/LinkFree/assets/135469703/5e6b2229-a9ee-49ad-9405-e7be2cd71235)

**CAN ALSO REPLACE VERCEL ANALYTICS IF WE ADD AN ADDITIONAL DATASET *we'll have to create helper functions that send the data though***

**Comment by CBID2** [comment]
Highlighting Non-coding contributions discussion
I recently read [this article](https://github.com/readme/featured/open-source-non-code-contributions). I think there’s more that can be done to show new members that Linkfree accepts non-coding contributions. Like a disclaimer or pin examples of merged non-code prs. By the way, it’s June 14th not July 14th(referring to the Office Hours title).

**Reply by eddiejaoude** [reply]
We do have non code contributions docs page, it would be good to add more information to this page https://linkfree.io/docs/community-resources

**Reply by CBID2** [reply]
Definitely! :) We should also put in the entrance of the Linkfree and EddieHub repo

**Reply by CBID2** [reply]
Hey @eddiejaoude! :) I raised the issue for this. Check #7581

**Comment by eddiejaoude** [comment]
Insights scores big difference between profiles?

![Screenshot 2023-06-14 at 20 23 36](https://github.com/EddieHubCommunity/LinkFree/assets/624760/2d14406f-01a9-40c9-98f6-6992c7c0b8bc)

**Comment by ikrishg** [comment]
# Improving profile speeds

## Solution: ISR

Incremental static regeneration will allow us to generate pages every X minutes rather than computing them every single time, this will improve performance by a millenia since websites will be mostly static

### Problem it poses

The page visits won't be logged properly

#### Solution to the problem of the solution: 1

Using Partytown, to send the page visit event later via the web worker

#### Solution to the problem of the solution: 2

Using tinybird to make the page visit be counted on middleware, this will also give us additional benefits like:

- Geolocation (we can make a map on the page!) and IP tracking (potentially count one via by a particular IP in a particular time frame?)
- We can also track that from where the traffic is coming and where it is going
- Realtime

**Reply by eddiejaoude** [reply]
Thanks for the suggestions 👍

**Comment by ikrishg** [comment]
# The link on buttons should be the link

## Why?

- SEO
- Copying it will be simpler

## How?

Currently we log link-visit events, logging click events instead should fix it 👍🏻

**Comment by eddiejaoude** [comment]
Switching to Typescript?

**Reply by ikrishg** [reply]
Yes typescript yes!

**Reply by ikrishg** [reply]
Also we have to move to `app/` directory
can be one big change 💪🏻

**Reply by ikrishg** [reply]
and we also had to implement radix-primitives 😱 so many major changes lmao

**Comment by ikrishg** [comment]
Moving linkfree docs to docs.linkfree.io to add a sidebar, search and overall make docs better

**Comment by ikrishg** [comment]
Using radix-ui to improve accessibility of tailwind accessibility and functionality

**Comment by eddiejaoude** [comment]
Call about https://github.com/EddieHubCommunity/LinkFree/issues/7065

Discussion issues...
- `reload` script to write to profile model once (currently many `findOneAndUpdate`)
   - other API calls have duplicate calls
- `next-auth` multiple oauth requests on login
- GitHub user location to use GH token to get over 60 requests per hour
- Improve logging https://github.com/EddieHubCommunity/LinkFree/issues/7438
- Associate profile to logged in user https://github.com/EddieHubCommunity/LinkFree/issues/7065

User case of association 
1. create json file, change GitHub username, after login profile will not load
2. create json file, login and associate account, after login show warning profile username does not match
3. use form to create profile, change GitHub username, show warning profile username does not match


Suggestions
- testimonials to have stars
- homepage to have more wow factor
- sharing profile link preview (OG tags)
-
